site/en/blog/prerender-pages/index.md

```diff
@@ -214,13 +247,13 @@ if (HTMLScriptElement.supports &&
 
 You can view a demo of Speculation Rules API prerendering, using JavaScript insertion, on [this demo page](https://prerender-demos.glitch.me/).
 
-#### Cancelling Speculation Rules
+### Cancelling speculation rules
 
-Removing Speculation Rules will result in the prerender being cancelled but, by the time this has happened, resources will likely have already been spent to initiate the prerender, so it is recommended not to prerender if there is a likelihood of needing to cancel the prerender.
+Removing speculation rules will result in the prerender being cancelled but, by the time this has happened, resources will likely have already been spent to initiate the prerender, so it is recommended not to prerender if there is a likelihood of needing to cancel the prerender.
 
-#### Speculation Rules and Content Security Policy
+### Speculation rules and Content Security Policy
 
-As Speculation Rules use a `<script>` element, even though they only contain JSON, they need to be included in the `script-src` [Content-Security-Policy](https://web.dev/csp/) if the site uses this—either using a hash or nonce. In future, a new `inline-speculation-rules` source will be supported in `script-src` allowing all `<script type="speculationrules">` elements on a page to be supported without the need of hashes or nonces.
+As speculation rules use a `<script>` element, even though they only contain JSON, they need to be included in the `script-src` [Content-Security-Policy](https://web.dev/csp/) if the site uses this—either using a hash or nonce. In future, a new `inline-speculation-rules` source will be supported in `script-src` allowing all `<script type="speculationrules">` elements on a page to be supported without the need of hashes or nonces.
```

Comment:
This is implemented from Chrome 110, so it's no longer really a future extension: https://chromestatus.com/feature/5182859125456896 .

Also it might be worth noting that you can use `strict-dynamic` and insert speculation rules from hashed/nonced scripts. (We had a bug but it's fix now!!!)

Reply:
Not quite fully implemented IMHO but made that clearer in https://github.com/GoogleChrome/developer.chrome.com/pull/6004/commits/6af7f216ec227ff6dae496c6b46e6396f00ffc7f

Reply:
Hmm, it should work on speculation rules included in the initial HTML...